language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/ModuleUtil.java

```diff
@@ -0,0 +1,265 @@
+                    aliasLastPart = "'" + aliasLastPart;
+                }
+                String aliasPart = Arrays.stream(aliasParts, 0, aliasParts.length - 1)
+                        .collect(Collectors.joining("."));
+                alias = aliasPart + "." + aliasLastPart;
+            } else {
+                if (CommonUtil.BALLERINA_KEYWORDS.contains(alias) && !preDeclaredLangLib) {
+                    alias = "'" + alias;
+                }
+            }
+            return orgName + "/" + alias;
+        }
+        return qualifiedModuleName;
+    }
+
+    /**
+     * Whether the package is already imported in the current document.
+     *
+     * @param context completion context
+     * @param pkg     Package to be evaluated against
+     * @return {@link Optional}
+     */
+    public static Optional<ImportDeclarationNode> matchingImportedModule(CompletionContext context,
+                                                                         LSPackageLoader.PackageInfo pkg) {
+        String name = pkg.packageName().value();
+        String orgName = pkg.packageOrg().value();
+        Map<ImportDeclarationNode, ModuleSymbol> currentDocImports = context.currentDocImportsMap();
+        return currentDocImports.keySet().stream()
+                .filter(importPkg -> importPkg.orgName().isPresent()
+                        && importPkg.orgName().get().orgName().text().equals(orgName)
+                        && ModuleUtil.getPackageNameComponentsCombined(importPkg).equals(name))
+                .findFirst();
+    }
+
+    /**
+     * Whether the package is already imported in the current document.
+     *
+     * @param context service operation context
+     * @param orgName organization name
+     * @param modName module name
+     * @return {@link Optional}
+     */
+    public static Optional<ImportDeclarationNode> matchingImportedModule(DocumentServiceContext context, String orgName,
+                                                                         String modName) {
+        Map<ImportDeclarationNode, ModuleSymbol> currentDocImports = context.currentDocImportsMap();
+        return currentDocImports.keySet().stream()
+                .filter(importPkg -> (importPkg.orgName().isEmpty()
+                        || importPkg.orgName().get().orgName().text().equals(orgName))
+                        && ModuleUtil.getPackageNameComponentsCombined(importPkg).equals(modName))
+                .findFirst();
+    }
+
+    /**
+     * Get the package name components combined.
+     *
+     * @param importNode {@link ImportDeclarationNode}
+     * @return {@link String}   Combined package name
+     */
+    public static String getPackageNameComponentsCombined(ImportDeclarationNode importNode) {
+        return importNode.moduleName().stream()
+                .map(Token::text)
+                .collect(Collectors.joining("."));
+    }
+
+    private static String getQualifiedModuleName(Module module) {
```

Comment:
Shall we add a javadoc here too?

Reply:
Not added, since this is a private method.